src/main/java/org/sagebionetworks/web/client/widget/ReactComponent.java

```diff
@@ -175,9 +176,17 @@ protected void onLoad() {
   protected void onUnload() {
     super.onUnload();
     if (root != null) {
-      root.unmount();
+      // Asynchronously schedule unmounting the root to allow React to finish the current render cycle.
+      // https://github.com/facebook/react/issues/25675
+      Timer t = new Timer() {
+        @Override
+        public void run() {
+          root.unmount();
+          root = null;
+        }
+      };
+      t.schedule(0);
```

Comment:
Fixes an issue where changing the type of a column would crash the app.

I think what was happening was the `EntityUpdatedEvent` was triggered by a React callback, which triggered a place reload, and this unload method ended up trying to remove the root DOM node at the same time that React was still using it.

By scheduling `unmount` instead of executing it synchronously, React gets a chance to finish the current render cycle, and now there's no longer a race condition.

Reply:
Sounds good.  And thank you for fixing all of the breaking changes that I've made (experimental mode, cookie banner, etc)!